op-e2e/custom_gas_token_bridge_test.go

```diff
@@ -0,0 +1,262 @@
+	require.NoError(t, err)
+	encodedUpgradeCall, err := proxyAdminABI.Pack("upgrade",
+		cfg.L1Deployments.SystemConfigProxy, storageSetterAddr)
+	require.NoError(t, err)
+
+	// Execute the upgrade SystemConfigProxy -> StorageSetter
+	tx, err = callViaSafe(t, cliqueSignerOpts, l1Client, proxyAdminOwner, cfg.L1Deployments.ProxyAdmin, encodedUpgradeCall)
+	waitForTx(t, tx, err, l1Client)
+
+	// Bind a StorageSetter to the SystemConfigProxy address
+	storageSetter, err := bindings.NewStorageSetter(cfg.L1Deployments.SystemConfigProxy, l1Client)
+	require.NoError(t, err)
+
+	// Use StorageSetter to clear out "initialize" slot
+	tx, err = storageSetter.SetBytes320(deployerOpts, [32]byte{0}, [32]byte{0})
+	waitForTx(t, tx, err, l1Client)
+
+	// Sanity check previous step worked
+	currentSlotValue, err := storageSetter.GetBytes32(&bind.CallOpts{}, [32]byte{0})
+	require.NoError(t, err)
+	require.Equal(t, currentSlotValue, [32]byte{0})
+
+	// Prepare calldata for SystemConfigProxy -> SystemConfig upgrade
+	encodedUpgradeCall, err = proxyAdminABI.Pack("upgrade",
+		cfg.L1Deployments.SystemConfigProxy, cfg.L1Deployments.SystemConfig)
+	require.NoError(t, err)
+
+	// Execute SystemConfigProxy -> SystemConfig upgrade
+	tx, err = callViaSafe(t, cliqueSignerOpts, l1Client, proxyAdminOwner, cfg.L1Deployments.ProxyAdmin, encodedUpgradeCall)
+	waitForTx(t, tx, err, l1Client)
+
+	version, err := systemConfig.Version(&bind.CallOpts{})
+	require.NoError(t, err)
+	t.Log("version: ", version)
+	// Reinitialise with existing initializer values but with custom gas token set
+	tx, err = systemConfig.Initialize(deployerOpts, owner,
+		overhead,
+		scalar,
+		batcherHash,
+		gasLimit,
+		unsafeBlockSigner,
+		resourceConfig,
+		batchInbox,
+		addresses)
+	waitForTx(t, tx, err, l1Client)
+
+	// Read Custom Gas Token and check it has been set properly
+	gpt, err := systemConfig.GasPayingToken(&bind.CallOpts{})
+	require.NoError(t, err)
+	require.Equal(t, cgtAddress, gpt.Addr)
+}
+
+func TestSetCustomGasToken(t *testing.T) {
+	InitParallel(t)
+
+	cfg := DefaultSystemConfig(t)
+
+	sys, err := cfg.Start(t)
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+
+	log := testlog.Logger(t, log.LevelInfo)
+	log.Info("genesis", "l2", sys.RollupConfig.Genesis.L2, "l1", sys.RollupConfig.Genesis.L1, "l2_time", sys.RollupConfig.Genesis.L2Time)
+
+	l1Client := sys.Clients["l1"]
+	deployerOpts, err := bind.NewKeyedTransactorWithChainID(cfg.Secrets.Deployer, cfg.L1ChainIDBig())
+	require.NoError(t, err)
+
+	// Deploy WETH, we'll use this as our custom gas token for the purpose of the test
+	wethAddress, tx, _, err := bindings.DeployWETH9(deployerOpts, l1Client)
+	require.NoError(t, err)
+	_, err = wait.ForReceiptOK(context.Background(), l1Client, tx.Hash())
+	require.NoError(t, err)
+
+	setCustomGasToken(t, cfg, sys, wethAddress)
+
+}
+
+func waitForTx(t *testing.T, tx *types.Transaction, err error, client *ethclient.Client) {
```

Comment:
fwiw we do have tooling for this elsewhere in op-e2e, these waits sometimes add flakes

Reply:
This is a thin wrapper around a util from the op-e2e/wait package, so this is at least not reinventing the wheel.